src/object.d

```diff
@@ -3241,6 +3241,226 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+int _adCmpT(T1, T2)(T1[] s1, T2[] s2)
```

Comment:
It might make sense to turn this into a stub that then calls the implementation with `U1` and `U2` to lower the number of template instantiations, if possible.